client/src/Context/AuthContext.tsx

```diff
@@ -1,77 +1,21 @@
 import { useState, createContext, useEffect, ReactNode } from "react";
 import { useNavigate } from "react-router-dom";
+import serverURL from "../config";
 import { getToken } from "../Utils/getToken";
 
-interface AuthContextType {
-  newUser: SignUp;
-  selectedFile: File | string;
-  submitImage: (e: React.MouseEvent<HTMLButtonElement>) => void;
-  signUp: (e: React.FormEvent) => void;
-  logIn: (e: React.FormEvent) => void;
-  setSelectedFile: (selectedFile: File | string) => void;
-  setNewUser: (newUser: SignUp) => void;
-  user: boolean;
-  setUser: (user: boolean) => void;
-  loginUser: Login;
-  setLoginUser: (loginUser: Login) => void;
-  logOut: () => void;
-  userProfile: User;
-  setUserProfile: (userProfile: User) => void;
-}
+//TODO decide if array should include object id or something else, then change 'any';
+//TODO display error messages
 interface Props {
   children: ReactNode;
 }
 
-//TODO sort interfaces in extra file?
-//TODO decide if array should include object id or something else, then change 'any';
-interface User {
-  email: string;
-  firstName: string;
-  image?: string;
-  isLoggedin?: boolean;
-  isAdmin?: boolean;
-  lastName: string;
-  starredCoworking?: Array<any>;
-  starredMentorship?: Array<any>;
-  starredShadowing?: Array<any>;
-  username: string;
-  _id: number;
-}
-
-interface SignUp extends User {
-  password: string;
-}
-interface LoginResult extends SignupResult {
-  isAuthenticated: boolean;
-}
-interface SignupResult {
-  accessToken: string;
-  message: string;
-  user: User;
-}
-interface ImageResult {
-  image: string;
-  message: string;
-}
-
-//TODO display error messages
-
 export const AuthContext = createContext<AuthContextType>(undefined!);
 
 export const AuthContextProvider: React.FC<Props> = ({ children }) => {
   const [user, setUser] = useState<boolean>(false);
   const [selectedFile, setSelectedFile] = useState<string | File>("");
-  const [userProfile, setUserProfile] = useState<User>({
-    firstName: "",
-    lastName: "",
-    username: "",
-    email: "",
-    image: "",
-    isAdmin: false,
-    isLoggedin: false,
-    _id: 0,
-  })
-  const [newUser, setNewUser] = useState<SignUp>({
+  const [userProfile, setUserProfile] = useState<User.User | null>(null);
```

Comment:
if you put here null, we have to check before using build in methods that userProfile isn't null. Might stay with variant before?

Reply:
Can you give a specific example when that would be a problem? We need this to be null so that conditional rendering can be applied properly (it's mostly going to be used with conditional rendering I think). I found it tricky with the previous notation (keys corresponding to empty strings). If you know of another elegant solution let me know - I found that to be the best. I find it not logical for this object to exist with keys and empty values if the user profile is not there before login or after refresh.

Reply:
Like you said it depends if we use conditional rendering. Initial states with typescript are often defined as empty strings. Maybe that clears what my thoughts were:
https://bobbyhadz.com/blog/react-usestate-with-null